### ISSUE TITLE
DBAL-1081: Paginator - Query Limit for SQL Server - SqlServerPlatform.php

### ISSUE THREAD

**Issue by doctrinebot** [doctrine/dbal#1017]
Jira issue originally created by user mgrajcarek:

Hi!
I have a problem with Query results limit when ordering by SUM of a field.

My query looks like this:

```
            SELECT ypk.name keyword, SUM(ypk.value) total
            FROM YpBundle:YellowPageKeyword ypk
              JOIN ypk.yellowpage yp
              JOIN yp.listing l
            WHERE l.customer = :customerId
              AND ypk.originDate >= :dateFrom
              AND ypk.originDate <= :dateTo
            GROUP BY ypk.nameCrc, ypk.name
            ORDER BY SUM(ypk.value) DESC
```

First I was catching following error:
message: `"[Syntax Error] line 0, col 395: Error: Expected known function, got 'SUM'"`
class: `Doctrine\ORM\Query\QueryException`

It only accourse if SUM is used in ORDER BY clause.

I have registered new class Sum which extends FunctionNode.

Now, query is build and executed but it has an error:

```
'SELECT * FROM 
     (SELECT y0*.name AS name_0, SUM(y0_.value) AS sclr_1, ROW_NUMBER() OVER (ORDER BY value) DESC) AS doctrine_rownum FROM yellowpage_keywords y0_ INNER JOIN yellowpages y1_ ON y0_.yellowpage_id = y1_.id INNER JOIN listings l2_ ON y1_.listing_id = l2_.id WHERE l2_.customer_id = ? AND y0_.origin_date >= ? AND y0_.origin_date <= ? GROUP BY y0_.name_crc, y0_.name) AS doctrine_tbl WHERE doctrine_rownum BETWEEN 1 AND 10 ORDER BY doctrine*rownum' with params ["111", "2013-01-01 00:00:00.000000", "2014-12-31 23:59:59.000000"]
```

The line : 
```ROW_NUMBER() OVER (ORDER BY value) DESC)
```
should look like 
```ROW*NUMBER() OVER (ORDER BY SUM(y0*.value) DESC)
```

In doModifyLimitQuery method I have modified:

```
                $pattern = sprintf('/%s\.%s\s<ins>(?:AS\s</ins>)?([<sup>,\s)]<ins>)/i', $column['table'], $column['column']);
```

to:

```
                $pattern = sprintf('/%s\.%s\s</ins>(?:AS\s<ins>)?([</sup>,\s)]</ins>)/i', str*replace('(', '\(', $column['table']), str*replace(')', '\)', $column['column']));
```
 
Now preg_match founds matching strings and `OVER` part of query is build correctly. 

I checked other issues about this problem (which are marked as already fixed)  and I have no idea why it's not working for me.

Thanks in advance!

**Comment by zeroedin-bill** [comment]
@deeky666 I think this can be closed as fixed, no?

**Comment by deeky666** [comment]
fixed by #818
